cranelift-codegen/src/redundant_reload_remover.rs

```diff
@@ -887,12 +887,13 @@ impl RedundantReloadRemover {
             cur: EncCursor::new(func, isa),
             reginfo: isa.register_info(),
             cfg: cfg,
-            state: &mut RedundantReloadRemover::new(),
+            state: self,
         };
         let mut total_regunits = 0;
         for rb in isa.register_info().banks {
             total_regunits += rb.units;
         }
+        ctx.state.clear();
```

Comment:
This is not needed, since this is called in the main Context clear() function, right?

Reply:
Mhm, I was not sure I could identify that clear call.  I just put it here to be safe, since it's the obvious place to put it.

Reply:
We should probably only keep of these two.